core/src/main/scripts/migrate_db.py

```diff
@@ -156,24 +156,32 @@ def check_reference_genome(portal_properties, cursor, force_migration):
     warnings = []
     try:
         sql_statement = """
-                           select GROUP_CONCAT(distinct NCBI_BUILD), count(distinct NCBI_BUILD), CANCER_STUDY_IDENTIFIER
+                           select NCBI_BUILD, count(NCBI_BUILD), CANCER_STUDY_IDENTIFIER
                            from mutation_event
                            join mutation on mutation.MUTATION_EVENT_ID = mutation_event.MUTATION_EVENT_ID
                            join genetic_profile on genetic_profile.GENETIC_PROFILE_ID = mutation.GENETIC_PROFILE_ID
                            join cancer_study on cancer_study.CANCER_STUDY_ID = genetic_profile.CANCER_STUDY_ID
-                           group by CANCER_STUDY_IDENTIFIER
+                           group by CANCER_STUDY_IDENTIFIER, NCBI_BUILD
                        """
         cursor.execute(sql_statement)
+        study_to_ncbi_to_count = {} # {cancer_study_identifier : {ncbi_build  : record_count}}
         for row in cursor.fetchall():
             retrieved_ncbi_build, ref_count, study = row
-            if int(ref_count) == 1:
-                retrieved_ncbi_build = list(set(retrieved_ncbi_build.split(',')))[0]
-                if retrieved_ncbi_build.upper() not in [x.upper() for x in ALLOWABLE_GENOME_REFERENCES]:
-                    msg = 'WARNING: Study %s contains mutation_event records with unsupported NCBI_BUILD value %s.'%(study, retrieved_ncbi_build)
-                    warnings.append(msg)
-            elif int(ref_count) > 1:
-                msg = 'WARNING: Study %s contains mutation_event records with %s NCBI_BUILD values (%s).'%(study, ref_count, retrieved_ncbi_build)
+            if study in study_to_ncbi_to_count:
+                study_to_ncbi_to_count[study][retrieved_ncbi_build] = ref_count
+            else:
+                study_to_ncbi_to_count[study] = {retrieved_ncbi_build : ref_count}
+        for study in study_to_ncbi_to_count:
+            ncbi_to_count = study_to_ncbi_to_count[study]
+            if len(ncbi_to_count) == 1:
```

Comment:
@sheridancbio , I would suggest to pull lines 176 ~ 183 out and into a sub routine let's call it validate_reference_genome_count
```
def validate_reference_genome_count(warnings, ncbi_to_count):
    if len(ncbi_to_count) == 1:
        for retrieved_ncbi_build in ncbi_to_count: # single iteration
            if retrieved_ncbi_build.upper() not in [x.upper() for x in ALLOWABLE_GENOME_REFERENCES]:
                msg = 'WARNING: Study %s contains mutation_event records with unsupported NCBI_BUILD value %s.'%(study, retrieved_ncbi_build)
                warnings.append(msg)
    elif len(ncbi_to_count) > 1:
        msg = 'WARNING: Study %s contains mutation_event records with %s NCBI_BUILD values {ncbi_build:record_count,...} %s.'%(study, len(ncbi_to_count), ncbi_to_count)
        warnings.append(msg)
     return warnings
```

Reply:
then you can iterate through study_to_ncbi_to_count and then call the new sub routine. 
```
        for study in study_to_ncbi_to_count:
            validate_reference_genome_count(warnings, study_to_ncbi_to_count[study])
```

Reply:
Thank you .. I have added the subroutine. I named it validate_reference_genome_values_for_study, because it validates both the count and the acceptability of the values. Also, the function does not need to return a value, because 'warnings' is a pass by reference argument so when the subroutine appends to that object, the modifications are visible in the caller.